kotlinx-coroutines-core/common/src/flow/operators/Transform.kt

```diff
@@ -34,6 +35,11 @@ public inline fun <T> Flow<T>.filterNot(crossinline predicate: suspend (T) -> Bo
 @Suppress("UNCHECKED_CAST")
 public inline fun <reified R> Flow<*>.filterIsInstance(): Flow<R> = filter { it is R } as Flow<R>
 
+/**
+ * Returns a flow containing only values that are instances of the given [klass].
+ */
+public fun <R : Any> Flow<*>.filterIsInstance(klass: KClass<R>): Flow<R> = filter { klass.isInstance(it) } as Flow<R>
```

Comment:
I was puzzled why such overload exists only on JVM in stdlib.

It seems like its existence just predates https://youtrack.jetbrains.com/issue/KT-14720, and it is no longer an issue. I suggest filing a corresponding ticket on stdlib

Reply:
Indeed, I didn't even notice! [Created a ticket](https://youtrack.jetbrains.com/issue/KT-54217/Add-IterablefilterIsInstanceklass-KClassR-extension-to-common-stdlib)